Rt1060 am457 bt support

MarkWangChinese opened the pull request:
with the three PRs: https://github.com/zephyrproject-rtos/zephyr/pull/31583, https://github.com/zephyrproject-rtos/zephyr/pull/35536 and this PR, the a2dp samples can run in the RT1060+ AW-AM457-D-uSD

MaureenHelm commented on drivers/serial/uart_mcux_lpuart.c:
```diff
@@ -356,7 +356,11 @@ static int mcux_lpuart_init(const struct device *dev)
-
+#if defined (CONFIG_BT_UART_ON_DEV_NAME)
+	if (strcmp(CONFIG_BT_UART_ON_DEV_NAME, dev->name) == 0) {
+		uart_api_config->flow_ctrl = UART_CFG_FLOW_CTRL_RTS_CTS;
```
Flow control should be implemented using the dts property, `hw-flow-control`, similar to `drivers/serial/uart_mcux.c`. We should not use `CONFIG_BT_UART_ON_DEV_NAME` here.

MarkWangChinese replied:
Thanks @MaureenHelm , I have upated it, please help check it again.

MaureenHelm commented on boards/arm/mimxrt1060_evk/pinmux.c:
```diff
@@ -144,6 +144,25 @@ static int mimxrt1060_evk_init(const struct device *dev)
 			    IOMUXC_SW_PAD_CTL_PAD_DSE(6));
+
+#if defined (CONFIG_BT_UART_ON_DEV_NAME)
+	if (strcmp(CONFIG_BT_UART_ON_DEV_NAME, "UART_3") == 0) {
```
This should use the dts property `hw-flow-control` instead of `CONFIG_BT_UART_ON_DEV_NAME`

MarkWangChinese replied:
Thanks @MaureenHelm , I have upated it, please help check it again.

MaureenHelm reviewed (changes requested):
Thanks @MarkWangChinese 

Please review and address the compliance check failure.

MarkWangChinese committed:
drivers: serial: NXP: Enable flow control for uart_mcux_lpuart

add flow_ctrl filed and give it initialized value
    based on hw_flow_control.
Initialize mcux lpuart based on flow_ctrl

Signed-off-by: Mark Wang <yichang.wang@nxp.com>

MaureenHelm reviewed (changes requested):
Code changes look good now, but you still need to address the compliance check failure:
```
Error: See docs.zephyrproject.org/latest/contribute/#commit-guidelines for more details

Commit 04723e69b2:
1: UC6 Body has no content, should at least have 1 line.

Commit 1a3d406117:
1: UC6 Body has no content, should at least have 1 line.
```

MarkWangChinese commented:
Hi @MaureenHelm, I have updated it, there are still some fails in buildkite/zephyr checking. It seems not related to my changes, please help confirm it. Thanks.

dleach02 commented on drivers/serial/uart_mcux_lpuart.c:
```diff
@@ -426,7 +427,9 @@ static const struct mcux_lpuart_config mcux_lpuart_##n##_config = {	\
 	.clock_subsys = (clock_control_subsys_t)DT_INST_CLOCKS_CELL(n, name),\
 	.baud_rate = DT_INST_PROP(n, current_speed),			\
-	IRQ_FUNC_INIT							\
+	IRQ_FUNC_INIT,							\
```
The error is related to the comma after IRQ_FUNC_INIT. I suspect the expansion on the particular build to get to this probably already includes a comma.

From log file:

 /workdir/zephyr/drivers/serial/uart_mcux_lpuart.c:430:15: error: expected expression before ',' token

  430 |  IRQ_FUNC_INIT,       \

      |               ^

../../../../../../../include/sys/util_internal.h:67:26: note: in definition of macro '__DEBRACKET'

   67 | #define __DEBRACKET(...) __VA_ARGS__

      |                          ^~~~~~~~~~~

../../../../../../../include/sys/util_internal.h:59:2: note: in expansion of macro '__GET_ARG2_DEBRACKET'

   59 |  __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)

      |  ^~~~~~~~~~~~~~~~~~~~

../../../../../../../include/sys/util_internal.h:54:2: note: in expansion of macro '__COND_CODE'

   54 |  __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)

      |  ^~~~~~~~~~~

../../../../../../../include/sys/util_macro.h:117:2: note: in expansion of macro 'Z_COND_CODE_1'

  117 |  Z_COND_CODE_1(_flag, _if_1_code, _else_code)

      |  ^~~~~~~~~~~~~

../../../../../../../include/devicetree.h:2211:2: note: in expansion of macro 'COND_CODE_1'

 2211 |  COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT), \

      |  ^~~~~~~~~~~

/workdir/zephyr/drivers/serial/uart_mcux_lpuart.c:421:2: note: in expansion of macro 'LPUART_MCUX_DECLARE_CFG'

  421 |  LPUART_MCUX_DECLARE_CFG(n, LPUART_MCUX_IRQ_CFG_FUNC_INIT)

      |  ^~~~~~~~~~~~~~~~~~~~~~~

/workdir/zephyr/drivers/serial/uart_mcux_lpuart.c:452:2: note: in expansion of macro 'LPUART_MCUX_INIT_CFG'

  452 |  LPUART_MCUX_INIT_CFG(n);

      |  ^~~~~~~~~~~~~~~~~~~~

zephyr/include/generated/devicetree_unfixed.h:9163:53: note: in expansion of macro 'LPUART_MCUX_INIT'

 9163 | #define DT_FOREACH_OKAY_INST_nxp_kinetis_lpuart(fn) fn(0)

      |                                                     ^~

../../../../../../../include/sys/util_internal.h:85:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_kinetis_lpuart'

   85 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__

      |                                    ^

/workdir/zephyr/drivers/serial/uart_mcux_lpuart.c:454:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY'

  454 | DT_INST_FOREACH_STATUS_OKAY(LPUART_MCUX_INIT)

      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~

MarkWangChinese replied:
Thanks, I switch the order of flow_ctrl and irq_config_func to put IRQ_FUNC_INIT in the last.

dleach02 commented:
@MarkWangChinese do you still want to pursue this PR? There are build errors that need to be addressed.

MarkWangChinese committed:
boards: mimxrt1060_evk: Enable lpuart3's flow control pin mux

configure lpuart3's CTS and RTS

Signed-off-by: Mark Wang <yichang.wang@nxp.com>
